moped-database/metadata/tables.yaml

```diff
@@ -3096,6 +3085,14 @@
 - table:
     name: moped_project_roles
     schema: public
+  array_relationships:
+    - name: moped_proj_personnel_roles
+      using:
+        foreign_key_constraint_on:
+          column: project_role_id
+          table:
+            name: moped_proj_personnel_roles
+            schema: public
```

Comment:
we weren't tracking this relationship and i tend to prefer that we just track everything, even when it's an unlikely use case.

Reply:
I thought about this after you brought it up in dev sync the other day, and I'm fully on-board that we should expose all that we can out of Hasura. I can't think of a downside because if we don't need to query it - we'd just never write a query for it. 

Definitely could be overlooking something but I agree that we should build in flexibility for ad-hoc stuff like you mentioned instead of it requiring new work.

Reply:
I think I've been the most questioning of the hasura-esque attitude of relate-everything-and-expose-everything. 

It comes from my instinct that the graphql endpoint is a part of the frontend application. The frontend needs data in a certain form, and the `graphql-engine` can provide the data in that form. It's the frontend, not the backend (database) which governs the graphql endpoint's shape, if you will.

That said, I do agree that if your endpoint can answer all queries, then it by definition can answer the queries you may want in your application. A well normalized database with relations accurately represented in the `graphql-engine` metadata is really everything you could ask for.

A final thought which really sells me and makes me want to lean into relate-everything-and-expose-everything mentality:

* A fixed graphql endpoint shape allows the "backend team" to work on the database as a service and the "frontend team" to work on the front-end and not have to communicate or agree or talk even. The graphql endpoint's shape is the API they each agree on.
* A "dynamic" graphql endpoint based on the DB schema encourages (requires?) the front-end developer experience to include understanding and manipulating the database and it's middleware. This is anti-knowledge-silo and the makings for better developers and system administrators.

Reply:
> It's the frontend, not the backend (database) which governs the graphql endpoint's shape, if you will.
I think that's a very good point, and I think the fact that Hasura doesn't expose relationships by default hints that they have the same philosophy about shaping the API.

But the idea of a "fixed graphql endpoint" captures my sentiment so much better than i could articulate. it's nice to just know what you're gonna get when you ask Hasura for data. Particularly because we control access to the API through auth and permissions, it feels like the best approach to me.

Reply:
thank you both for your insight! 🙌